src/main/java/com/SAS/User/UserController.java

```diff
@@ -16,14 +17,34 @@ public UserController() {
         this.globalPrivileges = Privileges.getInstance();
     }
 
+    /**
+     * The function returns new guest
+     * @return
+     */
+    public User createGuest() {
+        return new Guest();
+    }
+
     /**
      * The function receives userName, password, fullName and user type and creates a user
      * @param userName
      * @param password
      * @param fullName
      */
-    public User createUser(String userName, String password, String fullName, UserType type, boolean approval){
-        User newUser = new Registered(userName, password, fullName);
+    public User createUser(String userName, String password, String fullName, UserType type, boolean approval, User newUser){
+         //check user parameters
+         if (!validateDetails(userName, password, fullName, type)) {
+             System.out.println("The parameters aren't valid. Pleas enter the details again.");
+             return newUser;
+         }
+
+         //checks if the userName already exists
+         if (crud.isUserNameExist(userName)) {
+             System.out.println("The userName already exist. Please enter your details again.");
```

Comment:
update this because now CRUD is static and you don't need instance of it to use it's functions

Reply:
I'll fix it.